src/lib/core/WeaveMessageLayer.cpp

```diff
@@ -2488,5 +2489,32 @@ void WeaveMessageLayer::GetPeerDescription(char * buf, size_t bufSize, const Wea
         msgInfo->InCon);
 }
 
+/**
+ * @brief
+ *   Generate random Weave device Id.
+ *
+ * @details
+ *   This function generates 64-bit locally unique Weave device Id. This function uses cryptographically strong
+ *   random data source to guarantee uniqueness of generated value. Note that bit 56 of the generated Weave
+ *   device Id is set to 1 to indicate that generated Weave device Id is locally (not globally) unique.
+ *
+ * @param deviceId                      A reference to the 64-bit Weave device Id.
+ *
+ * @retval  #WEAVE_NO_ERROR             If Weave device Id was successfully generated.
+ */
+NL_DLL_EXPORT WEAVE_ERROR GenerateWeaveDeviceId(uint64_t & deviceId)
+{
+    WEAVE_ERROR err;
+    enum { kMask_WeaveDeviceIdLocallyUnique = 0x0100000000000000ULL };
```

Comment:
Is there ever a need for someone to introspect such an ID to determine if it is locally unique? If so, this enumeration should be in a public header file.

Reply:
These enumeration declaration was moved to the header file.